lib/functions/transfer.js

```diff
@@ -82,9 +82,31 @@ class Transfer extends AsyncGeneratorFunction {
                         headers: Object.assign({
                             [HTTP.HEADER.CONTENT_LENGTH]: blob.size,
                             [HTTP.HEADER.CONTENT_TYPE]: transferPart.metadata.contentType
-                        }, transferPart.headers)
+                        }, transferPart.targetHeaders)
                     });
                 }, this.options);
+            } else if (targetUrl && isFileProtocol(targetUrl) && transferPart.source.url) {
+                await retry(async () => {
+                    const totalSize = transferPart.metadata.contentLength;
+                    const response = await streamGet(transferPart.source.url, {
+                        headers: Object.assign({
+                            [HTTP.HEADER.RANGE]: `${HTTP.RANGE.BYTES}=${contentRange.low}-${contentRange.high}`
+                        }, transferPart.sourceHeaders)
+                    });
+                    const contentLength = response.headers.get(HTTP.HEADER.CONTENT_LENGTH);
+
+                    // there have been cases where the server does not honor the range header.
+                    // since we'll be reading the entire response into memory, protect against accidentally
+                    // reading a very large amount of data into memory. For example, if the server responds with
+                    // an entire 1GB file even though we only requested a portion of that file, avoid reading
+                    // the whole 1GB into memory.
+                    const actualContentDifference = contentLength - contentRange.length;
```

Comment:
Headers sometimes are string types. I would consider converting contentLength to int to ensure it's a numeric value.
```suggestion
                    const actualContentDifference = Number.parseInt(contentLength, 10) - contentRange.length;
```